src/config/load.c

```diff
@@ -198,75 +198,80 @@ cfgLoadUpdateOption(void)
     // If archive retention is valid for the command, then set archive settings
     if (cfgOptionValid(cfgOptRepoRetentionArchive))
     {
-        // For each possible repo, check and adjust the settings as appropriate
-        for (unsigned int optionIdx = 0; optionIdx < cfgOptionGroupIdxTotal(cfgOptGrpRepo); optionIdx++)
+        // !!! NEED CHECK HERE THAT IF ARCHIVE RETENTION IS 0 THEN ARCHIVE-CHECK MUST BE ENABLED WHEN VALID
```

Comment:
In that case, it makes sense to require archive-check to be enabled imo :+1:

Reply:
Yes, but this check needs to be done *across hosts* so that makes it pretty complicated. In other words we need to make sure that archive-check is enabled on the *repo host* when repo-retention-archive is enabled on the *pg host*. I don't see any obvious way to do this.